.github/workflows/tests.yml

```
@@ -429,8 +481,10 @@ jobs:
 
 
   portdb:
-    if: ${{ !failure() && !cancelled() }} # Allow previous steps to be skipped, but not fail
-    needs: linting-done
+    if: ${{ !failure() && !cancelled() && needs.changes.outputs.linting == 'true' }} # Allow previous steps to be skipped, but not fail
```

It isn't immediately obvious why we checking `linting` here instead of like `trial` or `integration`:

```suggestion
    # Allow previous steps to be skipped, but not fail. The linting changes checks scripts-dev/*
    # and other directories related to portdb, so re-use it here.
    if: ${{ !failure() && !cancelled() && needs.changes.outputs.linting == 'true' }}
```